### PR TITLE
scripts: Move SafeStruct to VulkanObject

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Started to move SafeStruct out, plan to do the Utils and Source as well (either in this PR or follow up)

### CODE REVIEW

File: layers/vulkan/generated/vk_safe_struct.h [hunk 1]
```diff
@@ -194,17 +195,17 @@ struct safe_VkDeviceCreateInfo {
...
     uint32_t waitSemaphoreCount;
-    VkSemaphore* pWaitSemaphores{};
+    VkSemaphore* pWaitSemaphores;
     const VkPipelineStageFlags* pWaitDstStageMask{};
     uint32_t commandBufferCount;
-    VkCommandBuffer* pCommandBuffers{};
+    VkCommandBuffer* pCommandBuffers;
     uint32_t signalSemaphoreCount;
-    VkSemaphore* pSignalSemaphores{};
+    VkSemaphore* pSignalSemaphores;
```
Comment (juan-lunarg) [inline layers/vulkan/generated/vk_safe_struct.h:207]: Minor point I've noticed. Some pointers used to default initialize to nullptr. Now they don't. Most likely not an issue.

However, my preference would be that all variables are default initialized via `{}`

Comment (juan-lunarg) [reply layers/vulkan/generated/vk_safe_struct.h:207]: I realize that this shouldn't matter if the codegen works properly. But in the case it doesn't work properly it's nice to have a stable default like nullptr to default to.

Comment (spencer-lunarg) [reply layers/vulkan/generated/vk_safe_struct.h:207]: agree, will fix

Comment (spencer-lunarg) [reply layers/vulkan/generated/vk_safe_struct.h]: fixed, and was able to re-write the code cleaner now taking a second look at why this was happening

File: layers/vulkan/generated/vk_safe_struct.h [hunk 2]
```diff
@@ -71,6 +71,7 @@ struct safe_VkImageMemoryBarrier {
     uint32_t dstQueueFamilyIndex;
     VkImage image;
     VkImageSubresourceRange subresourceRange;
```
Comment (juan-lunarg) [inline layers/vulkan/generated/vk_safe_struct.h:73]: FFTI I think all members of the struct should be initialized with {}. Not just the pointers.

**Review by juan-lunarg — APPROVED** [review]

File: layers/vulkan/generated/vk_safe_struct.h (hunk 2 above)
Comment (spencer-lunarg) [reply layers/vulkan/generated/vk_safe_struct.h:73]: well the nice thing now, anyone should be easy to fix this now

**Pull request merged**